src/handlers/expiry/ExpiryDateStrategyFactory.ts

```diff
@@ -0,0 +1,81 @@
+import expiryMapping from "../../assets/strategy-mapping.json";
+import { IExpiryDateStrategy } from "./IExpiryDateStrategy";
+import { TestTypeForExpiry } from "../../models/TestTypeforExpiry";
+import { NoImplementationStrategy } from "./strategies/NoImplementationStrategy";
+import { PsvDefaultExpiryStrategy } from "./strategies/PsvDefaultExpiryStrategy";
+import { PsvMostRecentExpiryStrategy } from "./strategies/PsvMostRecentExpiryStrategy";
+import { PsvRegistrationAnniversaryStrategy } from "./strategies/PsvRegistrationAnniversaryStrategy";
+import { HgvTrlAnnualTestStrategy } from "./strategies/HgvTrlAnnualTestStrategy";
+import { HgvTrlMostRecentExpiryStrategy } from "./strategies/HgvTrlMostRecentExpiryStrategy";
+import { HgvTrlFirstTestStrategy } from "./strategies/HgvTrlFirstTestStrategy";
+import { DateProvider } from "./providers/DateProvider.js";
+import { VEHICLE_TYPE, ERRORS, EXPIRY_STRATEGY } from "../../assets/Enums";
+
+
+export interface ExpiryConfig {
+  isHistoryRequired: boolean;
+  isRegistrationRequired: boolean;
+  testTypes: string[];
+  strategy: string;
+}
+
+// TODO: add unit tests for Factory
```

Comment:
I think we did add unit tests for that so we can probably remove it

Reply:
removed comments